src/txmempool.cpp

```diff
@@ -980,26 +980,25 @@ const CTxMemPool::setEntries & CTxMemPool::GetMemPoolChildren(txiter entry) cons
 
 CFeeRate CTxMemPool::GetMinFee(size_t sizelimit) const {
     LOCK(cs);
-    if (!blockSinceLastRollingFeeBump || rollingMinimumFeeRate == 0)
-        return CFeeRate(llround(rollingMinimumFeeRate));
```

Comment:
I don't feel entirely happy that a bug fix is part of a mostly refactoring commit. Would you mind to split up those bug fixes to a separate (minimal) commit and pull request?

Reply:
No, I think that is a great idea. Should help me with digging as well. Will do